src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs

```diff
@@ -5009,19 +5009,22 @@ override public Task<T> GetFieldValueAsync<T>(int i, CancellationToken cancellat
             IDisposable registration = null;
             if (cancellationToken.CanBeCanceled)
             {
-                registration = cancellationToken.Register(s => ((SqlCommand)s).CancelIgnoreFailure(), _command);
+                registration = cancellationToken.Register(SqlCommand.s_cancelIgnoreFailure, _command);
```

Comment:
Can we summarize it in a single line? something like:
`IDisposable registration = cancellationToken.CanBeCanceled? cancellationToken.Register(SqlCommand.s_cancelIgnoreFailure, _command) : (IDisposable) null;`

Reply:
We could but there is no reason to do so. It adds a useless null cast and hides the optionality of the statement by making it look like a single assignment when it's optional not a choice between two values.